frog/imports/client/App/TopBar.js

```diff
@@ -112,36 +111,47 @@ class LogoutMenu extends React.Component<*, *> {
 
 class TopBarController extends React.Component<
   { classes: any, history: any, location: any },
-  {}
+  { location: string }
 > {
   routes = [
-    { name: 'Graph Editor', to: '/teacher/graph' },
-    { name: 'Sessions', to: '/t' },
+    { name: 'Graph Editor', to: ['/teacher/graph'] },
+    { name: 'Sessions', to: ['/teacher/orchestration', '/t'] },
     {
       name: 'Activity Preview',
-      to: '/teacher/preview'
+      to: ['/teacher/preview']
     }
   ];
 
-  value = '/teacher/preview';
-
   constructor(props) {
     super(props);
-    const found = this.routes.filter(
-      route => props.location.pathname.indexOf(route.to) !== -1
-    )[0];
-
-    if (found !== undefined) {
-      this.value = found.to;
-    }
+    const found = this.matchLocationPrefix();
+    this.state = {
+      location: found?.name || 'Graph Editor'
+    };
   }
 
+  matchLocationPrefix = () => {
+    const split = this.props.location.pathname.split('/');
+    return this.routes.find(
+      route =>
+        !!route.to.find(to =>
+          to
+            .split('/')
+            .reduce(
+              (acc, word, idx) => acc && (!split[idx] || split[idx] === word),
+              true
+            )
+        )
+    );
+  };
+
   handleChange = (event, value) => {
-    this.value = value;
+    this.setState({ location: value });
   };
 
   render() {
-    const { classes } = this.props;
+    const { classes, location } = this.props;
```

Comment:
`location` is defined in `this.props` AND in `this.state`, this could be a bug. Recommend changing the variable name `location` inside `this.state` to avoid confusion

Reply:
Made suitable changes